src/symbolic/symbolic_memory_make.ml

```diff
@@ -73,56 +73,65 @@ module Make (Backend : M) = struct
 
   let load_8_u m a =
     let open Symbolic_choice_without_memory in
-    let+ a = must_be_valid_address m.data a in
+    let+ a = must_be_valid_address m.data a 1 in
     let v = Backend.loadn m.data a 1 in
     match Smtml.Expr.view v with
     | Val (Num (I8 i)) -> Symbolic_value.const_i32 (Int32.of_int i)
     | _ -> Smtml.Expr.cvtop (Ty_bitv 32) (Zero_extend 24) v
 
   let load_16_s m a =
     let open Symbolic_choice_without_memory in
-    let+ a = must_be_valid_address m.data a in
+    let+ a = must_be_valid_address m.data a 2 in
     let v = Backend.loadn m.data a 2 in
     match Smtml.Expr.view v with
     | Val (Num (I32 i16)) -> Symbolic_value.const_i32 (Int32.extend_s 16 i16)
     | _ -> Smtml.Expr.cvtop (Ty_bitv 32) (Sign_extend 16) v
 
   let load_16_u m a =
     let open Symbolic_choice_without_memory in
-    let+ a = must_be_valid_address m.data a in
+    let+ a = must_be_valid_address m.data a 2 in
     let v = Backend.loadn m.data a 2 in
     match Smtml.Expr.view v with
     | Val (Num (I32 _)) -> v
     | _ -> Smtml.Expr.cvtop (Ty_bitv 32) (Zero_extend 16) v
 
+  (* TODO: Smtml should do this. Make call to Expr.simplify *)
+  let try_to_remove_extract v =
+    match Smtml.Expr.view v with
+    | Extract ({ node = Concat (({ node = Ptr _; _ } as ptr), _); _ }, 8, 4) ->
+      ptr
+    | Extract ({ node = Concat (_, ({ node = Ptr _; _ } as ptr)); _ }, 4, 0) ->
+      ptr
+    | _ -> v
```

Comment:
I'm fine about waiting for the next Smt.ml release, so that we don't forget about this piece of code. :sweat_smile: WDYT?

Reply:
Fine by me, we can leave this pending for now :smiley: